x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -47,164 +53,180 @@ interface Props {
+      query: { time }
+    },
+    {
+      icon: 'loggingApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showHostLogsLinkLabel',
+        { defaultMessage: 'Show host logs' }
+      ),
+      condition: hostName,
+      path: `/link-to/host-logs/${hostName}`,
+      query: { time }
+    },
+    {
+      icon: 'loggingApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showTraceLogsLinkLabel',
+        { defaultMessage: 'Show trace logs' }
+      ),
+      target: traceId,
+      hash: `/link-to/logs`,
+      query: { time, filter: `trace.id:${traceId}` }
+    },
+    {
+      icon: 'infraApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showPodMetricsLinkLabel',
+        { defaultMessage: 'Show pod metrics' }
+      ),
+      condition: podId,
+      path: `/link-to/pod-detail/${podId}`,
+      query: infraMetricsQuery
+    },
+    {
+      icon: 'infraApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showContainerMetricsLinkLabel',
+        { defaultMessage: 'Show container metrics' }
+      ),
+      condition: containerId,
+      path: `/link-to/container-detail/${containerId}`,
+      query: infraMetricsQuery
+    },
+    {
+      icon: 'infraApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showHostMetricsLinkLabel',
+        { defaultMessage: 'Show host metrics' }
+      ),
+      condition: hostName,
+      path: `/link-to/host-detail/${hostName}`,
+      query: infraMetricsQuery
+    }
+  ].map(({ icon, label, condition, path, query }, index) => ({
+    icon,
+    key: `infra-link-${index}`,
+    child: (
+      <InfraLink path={path} query={query}>
+        {label}
+      </InfraLink>
+    ),
+    condition
+  }));
 
-    const items = [
-      ...this.getInfraActions(),
-      <EuiContextMenuItem icon="discoverApp" key="discover-transaction">
-        <EuiFlexGroup gutterSize="s">
-          <EuiFlexItem>
-            <DiscoverTransactionLink transaction={transaction}>
-              {i18n.translate(
-                'xpack.apm.transactionActionMenu.viewSampleDocumentLinkLabel',
-                {
-                  defaultMessage: 'View sample document'
-                }
-              )}
-            </DiscoverTransactionLink>
-          </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiIcon type="popout" />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiContextMenuItem>
-    ];
+  const uptimeLink =
+    transaction && transaction.url
+      ? url.format({
+          pathname: chrome.addBasePath('/app/uptime'),
+          hash: `/?${fromQuery(
+            pick(
+              {
+                dateRangeStart: urlParams.rangeFrom,
+                dateRangeEnd: urlParams.rangeTo,
+                search: `url.domain:${transaction.url.domain}`
+              },
+              (val: string) => !!val
+            )
+          )}`
+        })
+      : null;
```

Comment:
What do you think about getting rid of the ternary and `pick`:

```js
const uptimeLink = url.format({
  pathname: chrome.addBasePath('/app/uptime'),
  hash: `/?${fromQuery({
    dateRangeStart: urlParams.rangeFrom,
    dateRangeEnd: urlParams.rangeTo,
    search: `url.domain:${idx(transaction, _ => _.url.domain)}`
  })}`
});
```

And below:
```js
condition: transaction && transaction.url
```

This way it's obvious what the condition for showing the link is when glancing over the list of items.

Reply:
`pick` is in there to prevent empty URL parameters ending up in the URL (any key in the object will be added to the URL, whether it's `null`, `undefined` or an empty string).

Agree w/ the other comment, looks cleaner (1 condition instead of 2).

Reply:
Yeah, I don't mind that but @ogupte mentioned that too so I'll give in :)